File: 2018/day06b.py

#
#
# For each point, find the region within manhattan 10000.
# That's a diamond.
# Store just the start,end for each line? 20000?
#
# Intersect all the diamonds.
# What's the area of the remainder?

# Find the distance from each point in the grid to each point in the 
#

# Start with 10x10.
import sys

test = (
    (1, 1),
    (1, 6),
    (8, 3),
    (3, 4),
    (5, 5),
    (8, 9)
)

live = tuple(tuple(int(p) for p in ln.split(' ')) for ln in open('day06.txt').readlines())

data = test
TARGET = 32
MAXD = 25

data = live
TARGET = 10000
MAXD = 500

def ham(x0,y0,x1,y1):
    return abs(x0-x1) + abs(y0-y1)

knt = 0
for y in range(-MAXD,MAXD):
    sys.stdout.write('%d\r' % y )
    sys.stdout.flush()
    for x in range(-MAXD,MAXD):
        sumh = 0
        for px,py in data:
            sumh += ham(x,y,px,py)
            if sumh >= TARGET:
                break
        if sumh < TARGET:
            print x,y
            knt += 1

print knt

File: 2018/day06b.cpp
#include <iostream>
#include <fstream>
#include <vector>

struct point_t {
    int x;
    int y;
};

int test[][2] = {
    {1, 1},
    {1, 6},
    {8, 3},
    {3, 4},
    {5, 5},
    {8, 9}
};

void readlive( std::vector<point_t> & inx )
{
    std::ifstream ifs( "day06.txt" );
    point_t pt;
    while( ifs )
    {
        ifs >> pt.x >> pt.y;
        if( !ifs ) break;
        inx.push_back( pt );
    }
}

int ham( int x0, int y0, int x1, int y1 )
{
    return abs(x1-x0)+abs(y1-y0);
}

#define TARGET 10000

int main()
{
    std::vector<point_t> live;

    readlive( live );

    int knt = 0;
    for( int y = -12000; y < 12000; y++ )
    {
        std::cout << y << '\r' << std::flush;
        for( int x = -12000; x < 12000; x++ )
        {
            int sumh = 0;
            for( auto && pt : live )
            {
                sumh += ham( x, y, pt.x, pt.y );
                if( sumh >= TARGET )
                    break;
            }
            if( sumh < TARGET )
            {
                std::cout << x << "," << y << "\n";
                knt += 1;
            }
        }
    }

    std::cout << "\n" << knt << "\n";
}
